annotator/sqlite_import_geonames.py

```diff
@@ -0,0 +1,133 @@
+import sys
+import os
+import csv
+import unicodecsv
+import sqlite3
+import time
+from StringIO import StringIO
+from zipfile import ZipFile
+from urllib import urlopen
+from get_database_connection import ANNIE_DB_PATH
+
+def parse_number(num, default):
+    try:
+        return int(num)
+    except ValueError:
+        try:
+            return float(num)
+        except ValueError:
+            return default
+
+GEONAMES_ZIP_URL = "http://download.geonames.org/export/dump/allCountries.zip"
+
+geonames_field_mappings = [
+    ('geonameid', 'text primary key'),
+    ('name', 'text'),
+    ('asciiname', 'text'),
+    ('alternatenames', None),
+    ('latitude', 'real'),
+    ('longitude', 'real'),
+    ('feature_class', 'text'),
+    ('feature_code', 'text'),
+    ('country_code', 'text'),
+    ('cc2', 'text'),
+    ('admin1_code', 'text'),
+    ('admin2_code', 'text'),
+    ('admin3_code', 'text'),
+    ('admin4_code', 'text'),
+    ('population', 'integer'),
+    ('elevation', None),
+    ('dem', None),
+    ('timezone', None),
+    ('modification_date', None)
+]
+
+def read_geonames_csv():
+    print "Downloading geoname data from: " + GEONAMES_ZIP_URL
+    url = urlopen(GEONAMES_ZIP_URL)
+    zipfile = ZipFile(StringIO(url.read()))
+    print "done"
+    #Loading geonames data may cause errors without this line:
+    csv.field_size_limit(sys.maxint)
+    with zipfile.open('allCountries.txt') as f:
+        reader = unicodecsv.DictReader(f,
+            fieldnames=[k for k,v in geonames_field_mappings],
+            encoding='utf-8',
+            delimiter='\t',
+            quoting=csv.QUOTE_NONE)
+        for d in reader:
+            d['population'] = parse_number(d['population'], 0)
+            d['latitude'] = parse_number(d['latitude'], 0)
+            d['longitude'] = parse_number(d['longitude'], 0)
+            if len(d['alternatenames']) > 0:
+                d['alternatenames'] = d['alternatenames'].split(',')
+            else:
+                d['alternatenames'] = []
+            yield d
+
+def batched(array):
+    batch_size = 100
+    batch = []
+    for idx, item in enumerate(array):
+        batch.append(item)
+        batch_idx = idx % batch_size
+        if batch_idx == batch_size - 1:
+            yield batch
+            batch = []
+    yield batch
+
+def create_sqlite_db():
+    if os.path.exists(ANNIE_DB_PATH):

```

Comment:
We don't want to overwrite it because it take a long time to create. This kind of logic can be useful for making deployment scripts run faster. If someone wants to update the database they can delete the old one then rerun the script. If we ever need to force an update due to a schema change I'll start versioning the database.